train.py

```diff
@@ -89,8 +89,9 @@
     torch.cuda.set_device(device)
     master_process = ddp_rank == 0 # this process will do logging, checkpointing etc.
     seed_offset = ddp_rank # each process gets a different seed
-    assert gradient_accumulation_steps % torch.cuda.device_count() == 0
-    gradient_accumulation_steps //= torch.cuda.device_count()
+    assert ddp_world_size <= torch.cuda.device_count()
```

Comment:
Do we need this assert?

Actually was the original implementation here buggy for multi-node training? torch.cuda.device_count() would only return number of devices on one node.

Reply:
(the current assert is not a matter of need but potentially a bug, as the world size for e.g. 2 nodes of 8 gpus each would be 16, but i think device_count() would return 8? the docs don't spell this out but i think this is the case)